Framework/Algorithms/src/FFTSmooth2.cpp

```diff
@@ -263,26 +254,21 @@ void FFTSmooth2::Butterworth(int n, int order,
   filteredWS =
       API::WorkspaceFactory::Instance().create(unfilteredWS, 2, mx, my);
 
-  const Mantid::MantidVec &Yr = unfilteredWS->readY(0);
-  const Mantid::MantidVec &Yi = unfilteredWS->readY(1);
-  const Mantid::MantidVec &X = unfilteredWS->readX(0);
-
-  Mantid::MantidVec &yr = filteredWS->dataY(0);
-  Mantid::MantidVec &yi = filteredWS->dataY(1);
-  Mantid::MantidVec &xr = filteredWS->dataX(0);
-  Mantid::MantidVec &xi = filteredWS->dataX(1);
+  filteredWS->setSharedX(0, unfilteredWS->sharedX(0));
+  filteredWS->setSharedX(1, unfilteredWS->sharedX(0));
 
-  xr.assign(X.begin(), X.end());
-  xi.assign(X.begin(), X.end());
-  yr.assign(Yr.size(), 0);
-  yi.assign(Yr.size(), 0);
+  auto &uY0 = unfilteredWS->y(0);
+  auto &uY1 = unfilteredWS->y(1);
+  auto &Y0 = filteredWS->mutableY(0);
+  auto &Y1 = filteredWS->mutableY(1);
```

Comment:
I think it makes sense to keep the `r` and `i` in the variable names, since it makes it easier to understand what is happening mathematically (`r` and `i` refer to real part and imaginary part of the complex numbers).